File: leetsync/3321-type-of-triangle/type-of-triangle.go
func triangleType(nums []int) string {
	if !isTriangle(nums) {
		return "none"
	}
	if isEquilateral(nums) {
		return "equilateral"
	}
	if isIsosceles(nums) {
		return "isosceles"
	}
	return "scalene"
}

func isTriangle(nums []int) bool {
	return nums[0]+nums[1] > nums[2] &&
		nums[0]+nums[2] > nums[1] &&
		nums[1]+nums[2] > nums[0]
}

func isEquilateral(nums []int) bool {
	return nums[0] == nums[1] && nums[1] == nums[2]
}

func isIsosceles(nums []int) bool {
	return nums[0] == nums[1] || nums[1] == nums[2] || nums[0] == nums[2]
}

File: leetsync/3321-type-of-triangle/type-of-triangle.java
class Solution {
    public String triangleType(int[] nums) {
        int a = nums[0], b = nums[1], c = nums[2];
        if (a == b && b == c) {
            return "equilateral";
        } else if (((a == b) || (b == c) || (a == c)) && (a + b > c) && (b + c > a) && (a + c > b)) {
            return "isosceles";
        } else if (((a != b) && (b != c) && (a != c)) && (a + b > c) && (b + c > a) && (a + c > b)) {
            return "scalene";
        } else {
            return "none";
        }
    }
}